### ISSUE TITLE
User defined initialisation

### ISSUE THREAD

**Issue by yang-chen-2022** [eikehmueller/JaxMaterials#2]
Background
----------
Currently the Lippmann–Schwinger solvers (JAX and CUDA backends) initialise the strain field epsilon using the constant mean value epsilon_bar. For time-dependent / nonlinear workflows (e.g. fracture models) we often want to start the solver at each time step from the previous time-step solution (a fully spatially-varying epsilon), not from a constant mean.

Proposal
--------
Add an optional argument `epsilon_init` to the solver API that allows the user to provide a spatially varying initial strain field with shape (6, nx, ny, nz) (or broadcastable to that). 
Behavior:
- If epsilon_init is provided, use it as the solver initial iterate.
- If epsilon_init is None, preserve current behavior (initialise from epsilon_bar).

Compatibility / notes (by copilot, verified by YC)
- The default behavior (no epsilon_init) remains unchanged.
- CUDA path requires float32 arrays.
- For JAX gradients, if epsilon_init should be regarded as fixed (not a variable), call jax.lax.stop_gradient(epsilon_init) before passing it in.

**Comment by eikehmueller** [comment]
First implementation in 070a376. So far, only the JAX version allows passing in an optional parameter to initialise the strain. Still need to implement this in CUDA.

**Comment by eikehmueller** [comment]
To avoid rounding errors, it appears to be more sensible to pass epsilon_bar and delta_epsilon_init, which is either zero (= default value) or a field which averages to zero. This should avoid rounding errors when separating out the non-differentiable part. Could also check that delta_epsilon_init does indeed average to zero in the user interface.

**Comment by eikehmueller** [comment]
Created pull request #3

**Comment by yang-chen-2022** [comment]
It works well for my phase-field fracture model now. Reduced the number of iterations by a good factor.

**Comment by yang-chen-2022** [comment]
An IF statement was introduced in https://github.com/eikehmueller/JaxMaterials/blob/main/src/jaxmaterials/solver/lippmann_schwinger.py
line 131-143

This seems creating issue (TracerBoolConversionError) for my jitted phase-field solver:
https://github.com/yang-chen-2022/diffmat/blob/main/src/diffmat/fracture/solver_jit.py

Setting depsilon=None in for lippmann_schwingger seems solving the problem (as a temporary patch):
https://github.com/yang-chen-2022/diffmat/blob/main/src/diffmat/fracture/solve_one_step.py

You can reproduce the error by running this example:
https://github.com/yang-chen-2022/diffmat/blob/main/examples/example_pfm_test_jit.py
(set the spacing to larger values for speedup & also, change depsilon=depsilon in lippmann_schwinger).

**Comment by eikehmueller** [comment]
Yes, I ran into the same issue yesterday. The problem is the if-statement in the else-branch which verifies that the passed vector has zero average and raises an exception otherwise. 

This sort of error handling causes issues inside jitted functions. I think the error handlind should be removed.

See also issue #8 : if functions get transformed or jitted really strange things can happen. For example, it printed that the solver hadn't converged, even though it clearly had.

**Comment by eikehmueller** [comment]
Fix provided in 4aabe64
